packages/formatters/src/formatters.ts

```diff
@@ -336,11 +336,48 @@ export function formatEmoji<EmojiId extends Snowflake>(
 	animated?: boolean,
 ): `<:_:${EmojiId}>` | `<a:_:${EmojiId}>`;
 
-export function formatEmoji<EmojiId extends Snowflake>(
-	emojiId: EmojiId,
-	animated = false,
-): `<:_:${EmojiId}>` | `<a:_:${EmojiId}>` {
-	return `<${animated ? 'a' : ''}:_:${emojiId}>`;
+/**
+ * Formats an emoji id and name into a fully qualified emoji identifier.
+ *
+ * @param {FormatEmojiOptions} options - The options for formatting an emoji
+ */
+export function formatEmoji<EmojiId extends Snowflake, Name extends string>(
+	options: Required<FormatEmojiOptions<EmojiId, Name>>,
+): `<:${Name}:${EmojiId}>` | `<a:${Name}:${EmojiId}>`;
+
+export function formatEmoji<EmojiId extends Snowflake, Name extends string>(
+	emojiIdOrOptions: EmojiId | FormatEmojiOptions<EmojiId, Name>,
+	animated?: boolean,
+): `<:${string}:${EmojiId}>` | `<a:${string}:${EmojiId}>` {
+	const options =
+		typeof emojiIdOrOptions === 'string'
+			? {
+					id: emojiIdOrOptions,
+					animated: animated ?? false,
+				}
+			: emojiIdOrOptions;
+
+	const { id, animated: isAnimated, name: emojiName } = options;
+
+	return `<${isAnimated ? 'a' : ''}:${emojiName ?? '_'}:${id}>`;
+}
+
+/**
+ * The options for formatting an emoji.
```

Comment:
```suggestion
 * The options for formatting an emoji.
 *
 * @typeParam EmojiId - This is inferred by the supplied emoji id
 * @typeParam Name - This is inferred by the supplied name
```
Missed these type parameters, oops